src/wily/commands/build.py

```diff
@@ -157,7 +158,8 @@ def build(config: WilyConfig, archiver: Archiver, operators: List[Operator]) ->
 
                     # Aggregate metrics across all root paths using the aggregate function in the metric
                     # Note assumption is that nested dirs are listed after parent..
-                    for root in revision.tracked_dirs:
+                    dirs = {str(pathlib.Path(d)) for d in revision.tracked_dirs}
```

Comment:
This needs a comment to explain why there's a set comprehension with a str cast :-)